address-space-controller/src/main/java/io/enmasse/controller/StatusController.java

```diff
@@ -55,33 +57,61 @@ private InfraConfig parseCurrentInfraConfig(AddressSpace addressSpace) throws IO
         return type.getInfraConfigDeserializer().fromJson(addressSpace.getAnnotation(AnnotationKeys.APPLIED_INFRA_CONFIG));
     }
 
-    private void checkDeploymentsReady(AddressSpace addressSpace) {
+    private void checkComponentsReady(AddressSpace addressSpace) {
         try {
-            Set<String> readyDeployments = kubernetes.getReadyDeployments().stream()
-                    .map(deployment -> deployment.getMetadata().getName())
-                    .collect(Collectors.toSet());
-
             InfraConfig infraConfig = Optional.ofNullable(parseCurrentInfraConfig(addressSpace)).orElseGet(() -> getInfraConfig(addressSpace));
-            Set<String> requiredDeployments = infraResourceFactory.createInfraResources(addressSpace, infraConfig).stream()
-                    .filter(KubernetesHelper::isDeployment)
-                    .map(item -> item.getMetadata().getName())
-                    .collect(Collectors.toSet());
-
-            boolean isReady = readyDeployments.containsAll(requiredDeployments);
-            if (!isReady) {
-                Set<String> missing = new HashSet<>(requiredDeployments);
-                missing.removeAll(readyDeployments);
-                addressSpace.getStatus().setReady(false);
-                addressSpace.getStatus().appendMessage("Following deployments and statefulsets are not ready: " + missing);
-            }
+            List<HasMetadata> requiredResources = infraResourceFactory.createInfraResources(addressSpace, infraConfig);
+
+            checkDeploymentsReady(addressSpace, requiredResources);
+            checkStatefulSetsReady(addressSpace, requiredResources);
         } catch (Exception e) {
-            String msg = String.format("Error checking for ready deployments: %s", e.getMessage());
+            String msg = String.format("Error checking for ready components: %s", e.getMessage());
             log.warn(msg);
```

Comment:
We might need for the context for diagnostics purposes.  Suggest passing `e` to the `log.warn()`.